pkg/encoding/v1/encoding_test.go

```diff
@@ -144,9 +147,90 @@ func TestEnvVariable_UnmarshalYAML(t *testing.T) {
 	}
 }
 
+func TestService_UnmarshalYAML(t *testing.T) {
+
+	tests := []struct {
+		Name           string
+		ExpectedSucess bool
+		RawService     string
+		Service        *Service
+	}{
+		{
+			"Replica as positive int",
+			true, `
+name: frontend
+replicas: 3
+containers:
+- image: tomaskral/kompose-demo-frontend:test
+`,
+			&Service{
+				Name:     "frontend",
+				Replicas: goutil.Int32Addr(3),
+				Containers: []Container{
+					{
+						Image: "tomaskral/kompose-demo-frontend:test",
+					},
+				},
+			},
+		},
+
+		{
+			"Replica as 'string'",
+			false, `
+name: frontend
+replicas: notint
+containers:
+- image: tomaskral/kompose-demo-frontend:test
+`,
+			nil,
+		},
+
+		{
+			"Not giving any replica value it's an optional field",
+			true, `
+name: frontend
+containers:
+- image: tomaskral/kompose-demo-frontend:test
+`,
+			&Service{
+				Name: "frontend",
+				Containers: []Container{
+					{
+						Image: "tomaskral/kompose-demo-frontend:test",
+					},
+				},
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(fmt.Sprintf("Running test: %q", test.Name), func(t *testing.T) {
```

Comment:
There should be just a test name; go test tool will make the appropriate announcement on what is running